Geometry/TrackerNumberingBuilder/test/testTrackerPhiSort.cpp

```diff
@@ -40,120 +19,83 @@ namespace {
-    XY xy(r*std::cos(phi), r*std::sin(phi));
-    return xy;
+  double getPhi(XY const & xy) {
+    static const double pi2 = 2.*M_PI;
+    return xy.y >= 0 ? std::atan2(xy.y,xy.x) : pi2-std::atan2(-xy.y,xy.x);
   }
-}
-
 
-template<typename T>
-inline void crap(T t1, T t2) {
-  std::random_shuffle(t1,t2);
-}
+  template<typename V>
+  void printPhi(V const & v ) {
+    for(auto const& x : v) std::cout << getPhi(x) << " ";
+    std::cout << std::endl;
+  }
 
-template<typename V>
-void printPhi(V const & v ) {
-  std::transform(v.begin(),
-		 v.end(),
-		 std::ostream_iterator<double>(std::cout," "), 
-		 boost::bind(GetPhi(),_1)
-		 );
-		
-  std::cout << std::endl;
 }
 
-int main() {
+int main()
+{
   cppUnit::Dump a;
 
-  typedef std::vector<XY> Vector;
+  typedef std::vector<XY>     Vector;
+  typedef std::vector<Vector> Collection;
 
- typedef std::vector<Vector> Collection;
+  Collection original;
+  Collection shuffled;
+  Collection sorted;
 
-  Collection original, shuffled, sorted;
+  auto fromRP = [](double phi){ return XY(2*std::cos(phi), 2*std::sin(phi)); };
 
   // test1
   {
-    // ordered....
-    std::vector<double> phis; phis += 0.0, 0.5, 1.0, 1.5, 2.0, 2.5, 3., 3.1, 4.2, 5.2, 6.25;
-    // add an almost 0 (but not quite)
-    phis += GetPhi()(XY(1.E5,-1.));
-    original += Vector(phis.size());
-    std::transform(phis.begin(),phis.end(),original.back().begin(),boost::bind(fromRP,2.,_1));
+    // ordered.... last element is almost 0 (but not quite)
+    std::vector<double> phis {0.0, 0.5, 1.0, 1.5, 2.0, 2.5, 3., 3.1, 4.2, 5.2, 6.25, getPhi(XY(1.E5,-1.))};
+    original.push_back(Vector(phis.size()));
+    std::transform(phis.begin(),phis.end(),original.back().begin(),fromRP);
   }  // test1
   {
-    // ordered....
-    std::vector<double> phis; phis += 0.000001, 0.5, 1.0, 1.5, 2.0, 2.5, 3., 3.1, 4.2, 5.2, 6.25;
-    // add an almost 0 (is zero for the algo)
-    phis.insert(phis.begin(),(GetPhi()(XY(1.E10,-1.))));
-    original += Vector(phis.size());
-    std::transform(phis.begin(),phis.end(),original.back().begin(),boost::bind(fromRP,2.,_1));
-  }		 
+    // ordered.... first element is almost 0 (is zero for the algo)
+    std::vector<double> phis {getPhi(XY(1.E10,-1.)), 0.000001, 0.5, 1.0, 1.5, 2.0, 2.5, 3., 3.1, 4.2, 5.2, 6.25, };
+    original.push_back(Vector(phis.size()));
+    std::transform(phis.begin(),phis.end(),original.back().begin(),fromRP);
+  }
   // TEC tests???
   {
     // ordered....
-    std::vector<double> phis; phis += 0.0, 0.5, 3., 5.2, 6.25;
-    original += Vector(phis.size());
-    std::transform(phis.begin(),phis.end(),original.back().begin(),boost::bind(fromRP,2.,_1));
-  }		 
-		 
+    std::vector<double> phis {0.0, 0.5, 3., 5.2, 6.25};
+    original.push_back(Vector(phis.size()));
+    std::transform(phis.begin(),phis.end(),original.back().begin(),fromRP);
+  }
   {
     // ordered....
-    std::vector<double> phis; phis += -0.1683, -0.0561, +0.0000, +0.0561 +0.1683;
```

Comment:
I'd like to point out this nice example of why too much "boost magic" can be dangerous. Did the author intend to insert `+0.0561` and `+0.1683` as different elements and just forgot a comma, or was it really the idea to add the doubles and then insert the sum?

Reply:
forgot the comma